### ISSUE TITLE
Use compareWith validator instead of custom Validation rule

### ISSUE THREAD

**Issue by tom76dc** [CakeDC/users#428]
In UsersTable.php there is a custom validator to validate password against the password_confirm field

Ok, this adds no functionality. But i think it reads better with the 'new' compareWith validation rule

PS : This is my first github contribution, please be gentle :) . Can we also commit changes like this to the branch?

Regards, Tom

```
            $validator->add('password', [
                    'compare' =>    [
                                                'rule' => ['compareWith', 'password_confirm'],
                                                'message' => __d('CakeDC/Users', 'Your password does not match your confirm password. Please try again'),
                                                'on' => ['create', 'update'],
                                                'allowEmpty' => false   
                                                ]                       
                    ]);
```

**Comment by steinkel** [comment]
I agree, let's clean this up a bit...

**Comment by steinkel** [comment]
closed per #606
